proposals/nnnn-distributed-actors.md

```diff
@@ -0,0 +1,1621 @@
+
+This proposal is ABI additive.
+
+**TODO:** ???
+
+## Effect on API resilience
+
+**TODO:** ???
+
+## Changelog
+
+- Remove `init(transport:)` and base the design around synthesis of the lifecycle calls into existing designated initializers
+- Remove `init(resolve:using:)` and replace it with a static resolve factory function which works much better with the DI challanges that such remote instance might be facing
+- Change `struct ActorAddress` to `protocol ActorIdentity` which allows for transports to customize and pick efficient representations of identities as apropriate for their underlying transport mechanisms
+  - These may remain URI-like structures, or highly optimized numeric representations (e.g. unique actor ID number)
+  - Makes implementation dependent on [SE-0309: Unlock existentials for all protocols](https://github.com/apple/swift-evolution/blob/main/proposals/0309-unlock-existential-types-for-all-protocols.md)
+
+
+# Appendix
+
+## Runtime implementation details
+
+This section of the proposal discusses some of the runtime internal details of how distributed functions and remote references are implemented. While not part of the semantic model of the proposal, it is crucial for the implementation approach to fit well into Swift's runtime.
+
+#### Remote `distributed actor` instance allocation
+
+Creating a proxy for an actor type is done using a special `resolve(id:using:)` factory function of any distributed actor. Internally, it invokes the transport's `resolve` function, which determines if the identity resolves to a known local actor managed by this transport, a remote actor which this transport is able to communicate with, or the identity cannot be resolved and the resolve will throw:
+
+```swift
+protocol DistributedActor { 
+    static func resolve<ID>(_ identity: ActorIdentity, using transport: ActorTransport) 
+      throws -> Self { 
+    	// ... synthesized ...
+    }
+}
+
+protocol ActorTransport { 
+  /// Resolve a local or remote actor address to a real actor instance, or throw if unable to.
+  /// The returned value is either a local actor or nil, signifying that a 'remote' actor should be created.
+  func resolve<Act>(_ identity: AnyActorIdentity, as actorType: Act.Type) 
+      throws -> Act? where Act: DistributedActor
+}
+```
+
+This function can only be invoked on specific actor types–as usual with static functions on protocols–and serves as a factory function for actor proxies of given specific type.
+
+Implementing the resolve function by returning `.resolved(instance)` allows the transport to return known local actors it is aware of. Otherwise, if it intends to proxy messages to this actor through itself it should return `.proxy`, instructing the constructor to only construct a partial "proxy" instance using the address and transport. The transport may also chose to throw, in which case the constructor will rethrow the error, e.g. explaining that the passed in address is illegal or malformed.
```

Comment:
`.resolved(instance)` and `.proxy`--- leftover pre-`Optional<Act>` design or something else?